### PR TITLE
Finetune training hyperparameter

### PR DESCRIPTION
**Pull request by shaydeci**
Added finetune training hyperparameter.

- Implemented for all models that we have pre-trained weights for.
- Added tests for each model.
- Added interface for supporting fine-tune.
- Extended our utility check for comparing model's weights to skip batch normalization layers.

@BloodAxe I know you had other suggestions, @ofrimasad suggested that it would be easier for users that are familiar with our mechanism of assigning seperate learning rates to implement finetune.

### CODE REVIEW

File: src/super_gradients/training/models/sg_module.py [hunk 1]
```diff
@@ -64,5 +64,16 @@ class to implement.
...
+         when calling Trainer.train().
+        For example:
+            def get_finetune_lr_dict(self, lr: float) -> Dict[str, float]:
+                return {"default": 0, "head": lr}
+
+        :param lr: float, learning rate for the part of the network to be tuned.
+        :return: learning rate mapping that can be used by
+         super_gradients.training.utils.optimizer_utils.initialize_param_groups
+        """
+        raise NotImplementedError("Finetune is not implemented for this model, it is required to" " implement get_finetune_lr_dict.")
```
Comment (BloodAxe) [inline src/super_gradients/training/models/sg_module.py:79]: Some code formatting issues: ` to" " imple`

File: src/super_gradients/training/utils/optimizer_utils.py [hunk 2]
```diff
@@ -111,6 +112,12 @@ def build_optimizer(net: nn.Module, lr: float, training_params) -> optim.Optimiz
             "initial_lr training hyperparameter (i.e initial_lr={'backbone': 0.01, 'default':0.1})",
             DeprecationWarning,
         )
+    if training_params.finetune and isinstance(net, SupportsFineTune):
```
Comment (BloodAxe) [inline src/super_gradients/training/utils/optimizer_utils.py:115]: Perhaps we should raise a warning if `if training_params.finetune` is True and model is not supporting this and not just silently continue?

**Comment by BloodAxe** [comment]
LGTM overall, just a few minor cosmetic touches (typos)

File: src/super_gradients/training/models/sg_module.py (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/training/models/sg_module.py]: 5321b2de50e9e66ff755d3663be919bdd50c112f fixed

File: src/super_gradients/training/utils/optimizer_utils.py (hunk 2 above)
Comment (shaydeci) [reply src/super_gradients/training/utils/optimizer_utils.py]: sure, 5321b2de50e9e66ff755d3663be919bdd50c112f

File: src/super_gradients/training/params.py [hunk 3]
```diff
@@ -81,6 +81,8 @@
         "options": None,  # A dictionary of options to pass to the backend.
         "disable": False,  # Turn torch.compile() into a no-op for testing
     },  # torch.compile options from https://pytorch.org/docs/stable/generated/torch.compile.html
+    "finetune": False  # Whether to freeze a fixed part of the model (supported only for models that implement
+    # get_finetune_lr_dict, see SgMpdule.get_finetune_lr_dict. Tailored for each model class.)
```
[the remark this answers is not in the document]
Comment (shaydeci) [reply src/super_gradients/training/params.py]: fixed 5321b2de50e9e66ff755d3663be919bdd50c112f

**Review by BloodAxe — APPROVED** [review]
LGTM
